qiskit/algorithms/minimum_eigen_solvers/vqe.py

```diff
@@ -210,6 +199,8 @@ def ansatz(self, ansatz: Optional[QuantumCircuit]):
                 self._ansatz_params = sorted(ansatz.parameters, key=lambda p: p.name)
             else:
                 self._ansatz_params = list(ansatz.parameters)
+        else:
+            self.ansatz = RealAmplitudes()
```

Comment:
See the comment about the optimizer, this would allow for a more straightforward workflow instead of calling the setter again from within the setter 😄